sql/src/main/java/io/crate/execution/engine/indexing/GroupRowsByShard.java

```diff
@@ -49,16 +49,22 @@
 
     private final RowShardResolver rowShardResolver;
     private final List<? extends CollectExpression<Row, ?>> expressions;
+    private final List<? extends CollectExpression<Row, ?>> sourceInfoExpressions;
     private final Function<String, TItem> itemFactory;
     private final Supplier<String> indexNameResolver;
     private final ClusterService clusterService;
     private final boolean autoCreateIndices;
+    private final BiConsumer<ShardedRequests, String> itemFailureConsumer;
+    private final Function<ShardedRequests, Boolean> possibleSourceUriProcessor;
 
     GroupRowsByShard(ClusterService clusterService,
                      RowShardResolver rowShardResolver,
                      Supplier<String> indexNameResolver,
                      List<? extends CollectExpression<Row, ?>> expressions,
+                     List<? extends CollectExpression<Row, ?>> sourceInfoExpressions,
                      Function<String, TItem> itemFactory,
+                     BiConsumer<ShardedRequests, String> itemFailureConsumer,
```

Comment:
Not sure if it's an improvement, but maybe this could be named `itemFailureRecorder` or `itemFailureTracker` to give some indication of what the consumer is supposed to be doing with the arguments it receives. The `Consumer` is already in the type anyway.